_resources/port1.0/group/python-1.0.tcl

```diff
@@ -65,7 +65,7 @@ default python.consistent_destroot yes
 
 proc python_get_version {} {
     if {[string match py-* [option name]]} {
-        return [string range [option subport] 2 3]
+        return [lindex [regexp -inline {py(?:py)?(\d+)-.*} [option subport]] 1]
```

Comment:
So this means that `${python.version}` is not necessarily one of the elements of `${python.versions}`? I don't think I like that.

The regex should be anchored and can lose the trailing `.*`. Why does the result need to be treated as a list?

Reply:
> So this means that `${python.version}` is not necessarily one of the elements of `${python.versions}`? I don't think I like that.

Yeah; I can see what you mean. I think the core problem here is that adding PyPy support sort of breaks the meaning of `python.version` and `python.versions`; prior to this change, different Python versions were all we could select, so the name made perfect sense. PyPy, however, is _not_ a Python version, but each release does have an associated Python version. Perhaps that's an argument in favour of a new version for the PortGroup? This would allow us to rename the variables.

> Why does the result need to be treated as a list?

The result of `regexp -inline` [is a list](https://tcl.tk/man/tcl8.4/TclCmd/regexp.htm#M13) 🙂 This seemed to me the simplest way of doing the extraction: the zeroth element of the list is the full match, and the first element is the part we're interested in. Is there a better way?